### DISCUSSION TITLE
[FEATURE] Support backup photos and videos that were uploaded on iCloud

### DISCUSSION THREAD

**Discussion by Sixth2538** [immich-app/immich#1682] [Feature Request]
**Describe the bug**
When backing up 30,000 photo and video assets from iPhone, uploading is stuck at after 3,000 uploads. Upload status will also show progress of uploading, but number of "Backup" photo does not increase.

Optimise iPhone storage option is selected on this device.

However, newest photos will get uploaded as normal. Once the new photos finish uploading, uploading progress will continue but older photo will not get backed up.

**Task List**

*Please complete the task list below. We need this information to help us reproduce the bug or point out problems in your setup. You are not providing enough info may delay our effort to help you.*

- [Y] I have read thoroughly the README setup and installation instructions.
- [Y] I have included my `docker-compose` file.
- [Y] I have included my redacted `.env` file.
- [Y] I have included information on my machine, and environment.

Docker Compose:

version: "3.8"

services:
  immich-server:
    image: altran1502/immich-server:release
    entrypoint: ["/bin/sh", "./start-server.sh"]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
    env_file:
      - .env
    environment:
      - NODE_ENV=production
    depends_on:
      - redis
      - database
    restart: always
    networks:
      - traefik-public     

  immich-microservices:
    image: altran1502/immich-server:release
    entrypoint: ["/bin/sh", "./start-microservices.sh"]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
    env_file:
      - .env
    environment:
      - NODE_ENV=production
    depends_on:
      - redis
      - database
    restart: always
    networks:
      - traefik-public

  immich-machine-learning:
    image: altran1502/immich-machine-learning:release
    entrypoint: ["/bin/sh", "./entrypoint.sh"]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
    env_file:
      - .env
    environment:
      - NODE_ENV=production
    depends_on:
      - database
    restart: always
    networks:
      - traefik-public

  immich-web:
    image: altran1502/immich-web:release
    entrypoint: ["/bin/sh", "./entrypoint.sh"]
    env_file:
      - .env
    restart: always
    networks:
      - traefik-public

  redis:
    container_name: immich_redis
    image: redis:6.2
    restart: always
    networks:
      - traefik-public

  database:
    container_name: immich_postgres
    image: postgres:14
    env_file:
      - .env
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
      PG_DATA: /var/lib/postgresql/data
    volumes:
      - [local bind volume]:/var/lib/postgresql/data
    restart: always
    networks:
      - traefik-public

  immich-proxy:
    container_name: immich_proxy
    image: altran1502/immich-proxy:release
    ports:
      - 2283:80
    logging:
      driver: none
    depends_on:
      - immich-server
    labels:
       - traefik.enable=true
       - traefik.docker.network=traefik-public
       - traefik.http.routers.photoview.rule=Host(`photo.xxx.xxx`)
       - traefik.http.routers.photoview.entrypoints=https
       - traefik.http.routers.photoview.tls=true
       - traefik.http.routers.photoview.tls.certresolver=le
    restart: always
    networks:
      - traefik-public

networks:
    traefik-public:
      external: true

Environmental Variables:

DB_HOSTNAME=immich_postgres
DB_USERNAME=xxx
DB_PASSWORD=xxx
DB_DATABASE_NAME=immich
REDIS_HOSTNAME=immich_redis
UPLOAD_LOCATION=[local bind volume]
JWT_SECRET=xxx
ENABLE_MAPBOX=true
MAPBOX_KEY=xxx
VITE_SERVER_ENDPOINT=https://photo.xxx.xxx/api

**To Reproduce**
1. Open Immich app on iOS
2. Click back up
3. Select "Recent (All)" album
4. Start backup

**Expected behavior**
Upload all 30,000 assets to server. Continue uploading older photo as shown on progress bar.

**Screenshots**
![IMG_1150](https://user-images.githubusercontent.com/111109779/184821905-8566f11c-f8b6-46a3-b5bc-724fe3a28ecc.png)

**System**
- Phone OS [iOS, Android]: `15.6`
- Server Version: `1.22.0`
- Mobile App Version: `1.22.0 build 42`

**Additional context**

**Comment by MalteKiefer** [comment]
I think you must disable the Optimize system.
The images must be on the device

**Comment by 20pins** [comment]
> I think you must disable the Optimize system.
> 
> The images must be on the device

I wonder how many other iOS issues are a result of this. I also have Optimize enabled. I currently have 3570 of 13k items uploaded.

**Comment by alextran1502** [comment]
If the assets are on iCloud, they won't be uploaded at this stage. I will look into this issue when I implemented Live Photo feature

**Reply by uninvitedguest** [reply]
Bringing back an old topic here, but I am wondering how things currently work on iOS (not my OS of choice).

My wife has 60k assets that show in her Recent album as a result of years of iCloud backups. I *don't* want what is stored on iCloud to be uploaded to Immich, I only want the recent, local photos uploaded.

It doesn't look like there are any "local only" albums that can be selected for backup in Immich, so I am curious what will happen if I select Recents...

**Reply by ScottEgan** [reply]
I'm no expert but I just went through this with my wife's 40k photos. Just selected Recents and all of them were uploaded eventually. 

You may have to look into this further if that is not what you want.

**Reply by ioslife** [reply]
@alextran1502, could you speak to how this works? Does Immich intelligently download the photos from iCloud so that we can get the highest quality available images? https://github.com/immich-app/immich/discussions/4267 also is asking about this

**Reply by alextran1502** [reply]
@ioslife I cannot speak for this as I haven't tested scenarios of phone using iCloud 😅 I don't use iCloud

**Reply by ioslife** [reply]
Ideally I'd like to not use iCloud either lol. I am going through my first upload now for Immich with my device. It has "Optimize iPhone Storage" enabled. I'll report my results in #4267 and let you know what happens.

**Comment by Sixth2538** [comment]
Thank you for the info and rapid development of the app. 

Seems like choosing optimizing storage is the root of the problem. Looking forward to see if this can be sorted out. Seems like it is by design to tie people down with the system.

I have tried various method to download my entire library and there really is no easy way to do it. If this feature can be implemented, it could be a game changer for many iOS users.

**Comment by esmondmissen** [comment]
Oddly enough I have optimise storage on and the backup is working fine. I've noticed it's a lot slower, but still works for me. I am on iOS 16 however.

**Comment by MalteKiefer** [comment]
How many photos do you have?

**Comment by esmondmissen** [comment]
About 17k, I definitely don’t have enough storage on my device to have them all locally.

**Comment by 2600box** [comment]
I am on 15.1 and no photos from my Photos.app show up in Immich. I have the icloud optimised storage turned on. 

Interesting if iOS16 is in fact the determining factor...

**Comment by Sixth2538** [comment]
Happy to report back that iOS 16 backup function is working and backing up all photo and video assets slowly. Even with optimised storage turned on.

**Comment by alextran1502** [comment]
@Sixth2538 WOOO! This is good news! I guess we can reduce additional code thanks to iOS 16 :)

**Comment by gorootde** [comment]
Any updates on this one? I am on 16.1.1 and still no photos show up at all (optimize storage enabled)

**Comment by jwinut** [comment]
I just started using immich. great product!

I also encounter this problem from iCloud Shared Album. I am subscribing to the albums and don't have those images on my iPhone (images not taken by me and I disable iCloud Photos on my device/ Shared Album enable).

I notice that the upload stuck at 100% on the same file name but when I Cancel the backup and start again, immich app backup 3-5 images before stuck at 100% again.

I kept smashing Cancel and Start Backup until my Shared Album is backed up (~100 images + some video).

my device: iPhone X
iOS 15.7.1

**Comment by alextran1502** [comment]
@jwinut There are some differences in the internal APIs of iOS in term of accessing the file system between version 15 and 16 I believe. Would it be possible for your phone to update to iOS 16?

**Comment by jwinut** [comment]
@alextran1502 it is possible. Would it help you with your development? if yes then I will.

**Comment by jwinut** [comment]
Updated to iOS 16.1.1. same behavior. need to cancel and start backup multiple times to complete remaining backup.

**Comment by bo0tzz** [comment]
THis is now supported

**Reply by tribey2** [reply]
Ah this is very timely, I'm new to Immich and have just set everything up today - but have found that anything which isn't on device (e.g. shared iCloud albums) get stuck when backing up.

I did find a workaround though - if you head to the shared album in question, select all photos, click the share button and then choose an app like 'Dropbox'. iOS will then pause and download all of the photos in that album directly onto the device ready for Dropbox (it indicates this on screen). Once that completes, cancel out of the dropbox upload and head back to Immich and then they all upload.

But it's great to hear that this is coming, which version? I'm on v1.91.4.
